### PR TITLE
Filter out collection types & get all columns for a primary-key

### PR DESCRIPTION
**Pull request by yarongilor**
fix(indexes MV): correct filtering of primary key an collections
    
            1. If no column found -> nemesis is skipped.
            2. collection-columns cannot be used for MV so they are filtered out.
            3. All columns of a primary-key should be included, when creating MV.
            4. Start a stopped node, in case of getting a nemesis exception.
 
Fixes errors encountered in add_remove_mv nemesis.
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
Not setting as ready-for-review until all related test cases pass.

**Comment by yarongilor** [comment]
Testing filtering passed ok.
```
< t:2023-04-19 15:29:21,941 f:common.py       l:1652 c:utils                p:DEBUG > Executing CQL 'CREATE TABLE t (id int, v int, v2 list<text>, PRIMARY KEY (id, v))' ...
< t:2023-04-19 15:29:21,941 f:common.py       l:1660 c:utils                p:DEBUG > Executing CQL 'CREATE TABLE t (id int, v int, v2 list<text>, PRIMARY KEY (id, v))' ...
< t:2023-04-19 15:29:22,267 f:common.py       l:1652 c:utils                p:DEBUG > Executing CQL 'CREATE MATERIALIZED VIEW t_by_v AS SELECT * FROM t WHERE v IS NOT NULL AND id IS NOT NULL PRIMARY KEY (v, id)' ...
< t:2023-04-19 15:29:22,268 f:common.py       l:1660 c:utils                p:DEBUG > Executing CQL 'CREATE MATERIALIZED VIEW t_by_v AS SELECT * FROM t WHERE v IS NOT NULL AND id IS NOT NULL PRIMARY KEY (v, id)' ...
< t:2023-04-19 15:29:59,979 f:nemesis.py      l:4323 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.AddRemoveMvNemesis: Start disruption at `2023-04-19 15:29:59.979066`
< t:2023-04-19 15:29:59,979 f:nemesis.py      l:1100 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.AddRemoveMvNemesis: Set current_disruption -> AddRemoveMv Node test-docker-yarongilor-db-node-dd4936ea-1 [172.17.0.7 | 172.17.0.7] (seed: False)
< t:2023-04-19 15:30:01,500 f:nemesis.py      l:4208 c:sdcm.nemesis         p:INFO  > sdcm.nemesis.AddRemoveMvNemesis: Stopping Scylla on node1
< t:2023-04-19 15:30:01,845 f:indexes.py      l:52   c:sdcm.utils.nemesis_utils.indexes p:DEBUG > res_list: [Row(column_name='id', type='int'), Row(column_name='v', type='int')]
< t:2023-04-19 15:30:01,846 f:indexes.py      l:54   c:sdcm.utils.nemesis_utils.indexes p:DEBUG > res_list column_names: ['id', 'v']
< t:2023-04-19 15:30:01,848 f:indexes.py      l:52   c:sdcm.utils.nemesis_utils.indexes p:DEBUG > res_list: [Row(column_name='v2', type='list<text>')]
< t:2023-04-19 15:30:01,849 f:indexes.py      l:54   c:sdcm.utils.nemesis_utils.indexes p:DEBUG > res_list column_names: ['v2']
< t:2023-04-19 15:30:01,849 f:indexes.py      l:58   c:sdcm.utils.nemesis_utils.indexes p:DEBUG > res_list column_names filter_out_collections: []
< t:2023-04-19 15:30:01,857 f:nemesis.py      l:4371 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.AddRemoveMvNemesis: AddRemoveMv Node test-docker-yarongilor-db-node-dd4936ea-1 [172.17.0.7 | 172.17.0.7] (seed: False) duration -> 1 s
< t:2023-04-19 15:30:01,857 f:nemesis.py      l:299  c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.AddRemoveMvNemesis: Update nemesis info with: {'start': 1681918199, 'end': 1681918201, 'duration': 1, 'node': 'Node test-docker-yarongilor-db-node-dd4936ea-1 [172.17.0.7 | 172.17.0.7] (seed: False)', 'subtype': 'skipped', 'skip_reason': "A supported column for creating MV is not found. nemesis can't run"}
```

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2324,7 +2322,8 @@ def toggle_table_gc_mode(self):
             new_gc_mode = GcMode.TIMEOUT
         new_gc_mode_as_dict = {'mode': new_gc_mode.value}
 
-        alter_command_prefix = 'ALTER TABLE ' if keyspace_table not in mview_ks_cfs else 'ALTER MATERIALIZED VIEW '
+        alter_command_prefix = 'ALTER TABLE ' if not is_cf_a_view(
```
Comment (yarongilor) [inline sdcm/nemesis.py:2325]: This line is not directly related to the main PR fix, but it was missing on a previous PR and be good to align with it.

**Review by fgelcer — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Comment by fgelcer** [comment]
@scylladb/qa-maintainers please merge and backport

**Comment by fgelcer** [comment]
thank you
